test/floki_test.exs

```diff
@@ -119,6 +132,41 @@ defmodule FlokiTest do
     ]
   end
 
+  # Floki.raw_html/2
+
+  test "raw_html" do
+    raw_html = Floki.parse(@basic_html) |> Floki.raw_html
+    assert raw_html == String.split(@basic_html, "\n") |> Enum.map(&(String.strip(&1))) |> Enum.join("")
+  end
+
+  test "raw_html (html with data attributes)" do
+    raw_html = Floki.parse(@html_with_data_attributes) |> Floki.raw_html
+    assert raw_html == String.split(raw_html, "\n") |> Enum.map(&(String.strip(&1))) |> Enum.join("")
+  end
+
+  test "raw_html (after find)" do
+    raw_html = Floki.parse(@basic_html) |> Floki.find("a") |> Floki.raw_html
+    assert raw_html == ~s(<a href="uol.com.br" class="bar"><span>UOL</span><img src="foo.png"/></a>)
+  end
+
+  # Floki.tag_attrs/1
```

Comment:
As I suggested to change those functions to private, you can remove the tests below.